include/beast/http/string_body.hpp

```diff
@@ -117,14 +120,14 @@ struct string_body
             {
                 body_.resize(len + n);
             }
-            catch(std::length_error const&)
+            catch(std::exception const&)
             {
                 ec = error::buffer_overflow;
                 return;
             }
-            ec.assign(0, ec.category());
             buffer_copy(boost::asio::buffer(
                 &body_[0] + len, n), buffers);
```

Comment:
if `std::distance(buffers.begin(), buffers.end()==1` then using `body_.append` is more efficient